apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/ServletApiAdvice.java

```diff
@@ -239,10 +239,26 @@ public static <HttpServletRequest, HttpServletResponse, ServletContext, ServletC
                     }
                 }
 
-                servletTransactionHelper.onAfter(transaction, t == null ? t2 : t, adapter.isCommitted(httpServletResponse), adapter.getStatus(httpServletResponse),
-                    overrideStatusCodeOnThrowable, adapter.getMethod(httpServletRequest), parameterMap, adapter.getServletPath(httpServletRequest),
-                    adapter.getPathInfo(httpServletRequest), contentTypeHeader, true
-                );
+                ServletContext servletContext = adapter.getServletContext(httpServletRequest);
+                String servletPath = adapter.getServletPath(httpServletRequest);
+                String pathInfo = adapter.getPathInfo(httpServletRequest);
+                if (servletPath == null || servletPath.length() == 0 && servletContext != null) {
```

Comment:
```suggestion
                if ((servletPath == null || servletPath.length() == 0) && servletContext != null) {
```

IINM, it would be equivalent to `if (servletPath == null || (servletPath.length() == 0 && servletContext != null))` otherwise.

Reply:
Good catch!